token/cli/src/main.rs

```diff
@@ -943,6 +943,21 @@ fn command_burn(
     })
 }
 
+
+fn check_mint(
+    config: &Config,
+    token: Pubkey,
+    ui_amount : f64,
```

Comment:
The UI amount can't be trusted totally, since an `f64` can't contain the same precision as a `u64`.  You'll have to convert the `ui_amount` into an `amount` and also get a fixed-point amount for the supply.  This means that you might have to move this check inside of `command_mint`